models/app.model.js

```diff
@@ -31,4 +31,48 @@ const fetchArticles = () => {
     })
 }
 
-module.exports = { readTopics, fetchArticlesbyId, fetchArticles }
\ No newline at end of file
+const fetchCommentsByArticleId = (article_id) => {
+    return db.query(`SELECT articles.article_id, comment_id, comments.votes, comments.created_at, comments.author, comments.article_id 
+    FROM articles 
+    LEFT JOIN comments ON articles.article_id = comments.article_id 
+    WHERE articles.article_id = $1`, [article_id])
+    .then(({rows}) => {
+        if (rows.length === 0) {
+            return Promise.reject({ status: 404, msg: 'not found' })
+        } else {
+            return db.query(`SELECT comments.comment_id, comments.votes, comments.created_at, comments.author, comments.body, comments.article_id 
+            FROM articles
+            LEFT JOIN comments ON articles.article_id = comments.article_id
+            WHERE comments.article_id = $1
+            ORDER BY created_at DESC;`, [article_id])
+            .then(({rows}) => {           
+                return rows
+            })
+        }
+    })     
```

Comment:
A further hint for this one. As mentioned previously in my feedback, there are two scenarios where we get empty arrays: no comments for a valid article or a non-existent article. We will need to check the article exists in some way and once that is done - only once it's done - should we continue on. Do you remember how we can handle multiple promises?